SignalMessaging/attachments/AttachmentApprovalViewController.swift

```diff
@@ -170,26 +164,18 @@ public class AttachmentApprovalViewController: OWSViewController {
         topToolbar.items = [cancelButton]
 
         // Bottom Toolbar
-        self.bottomToolbar = makeClearToolbar()
-        // Making a toolbar transparent requires setting an empty uiimage
-        bottomToolbar.setBackgroundImage(UIImage(), forToolbarPosition: .any, barMetrics: .default)
-        bottomToolbar.backgroundColor = UIColor.clear
-
-        let sendTitle = NSLocalizedString("ATTACHMENT_APPROVAL_SEND_BUTTON", comment: "Label for 'send' button in the 'attachment approval' dialog.")
-        let sendButton = UIBarButtonItem(title:  sendTitle,
-                                         style: .plain,
-                                         target: self,
-                                         action: #selector(sendPressed))
-        sendButton.tintColor = UIColor.white
+        let captioningToolbar = CaptioningToolbar()
+        captioningToolbar.captioningToolbarDelegate = self
+        self.bottomToolbar = captioningToolbar
+    }
 
-        let flexibleSpace = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
-        bottomToolbar.items = [flexibleSpace, sendButton]
+    override public var inputAccessoryView: UIView? {
+        self.bottomToolbar.layoutIfNeeded()
```

Comment:
Rather than observing keyboard - change notifications, we can use the inputAccessoryView/becomeFirstResponder combo. 

I'm hopeful that we can do something similar with the input toolbar in the ConversationView to get a nicer dismiss animations when leaving the view and also to apply the interactive dismiss when scrolling - but that's for another day.

Reply:
Let's discuss offline.